client/blocks/jitm/templates/modal.jsx

```diff
@@ -1,23 +1,60 @@
 import { Button, Guide } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
 import { useState } from 'react';
 import './modal-style.scss';
+import Inbox from './inbox.svg';
 import Visual from './plans-visual.svg';
 
 export default function ModalTemplate( {
-	featureClass,
 	CTA,
-	trackImpression,
-	message,
 	description,
+	disclaimer,
+	featureClass,
+	icon,
+	message,
 	onClick,
 	onDismiss,
 	title,
-	disclaimer,
+	trackImpression,
 } ) {
 	trackImpression && trackImpression();
 
 	// technically, non-dismissable jitms are authorable, however, that doesn't make any sense as a modal.
 	const [ isDismissed, setDismissed ] = useState( [] );
+	const translate = useTranslate();
+
+	const getModalImage = () => {
+		switch ( icon ) {
+			case 'embedded-inbox':
+				return Inbox;
+			default:
+				return Visual;
+		}
+	};
+
+	const getModalAltText = () => {
+		switch ( icon ) {
+			case 'embedded-inbox':
+				return translate( 'Embedded inbox', { textOnly: true } );
+			default:
+				return translate( 'list of available plans — premium, business, and ecommerce plans', {
```

Comment:
I would suggest the following text, so we don't have to update it when we launch new plans (it's already stale 😁):
```suggestion
				return translate( 'List of available WordPress.com plans', {
```